Sensor status now indicates whether the result were fetched from cache

mentisy opened the pull request:
According to feature #7 it was suggested to add the possibility to see whether sensor status was fetched from cache or if it was just run. This adds that feature.

ravage84 commented on src/Heartbeat/Sensor/Status.php:
```diff
@@ -127,4 +134,24 @@ public function getSeverity()
+     *
+     * @param bool $wasCached
+     */
+    public function setCheckWasCached(bool $wasCached): void {
```
I think we can't use ``void`` here because of PHP 7.0.0 BC https://github.com/orca-services/cakephp-heartbeat/blob/cakephp-3.x/composer.json#L37

mentisy replied:
Yeah, didn't think about the constraint. Removed it in a commit.

ravage84 commented on src/Heartbeat/Sensor.php:
```diff
@@ -86,11 +86,18 @@ protected function _getCachedStatus()
+        $cachedStatus = Cache::read($cacheKey, 'heartbeat');
 
-        return $cachedStatus;
+        if($cachedStatus !== false) {
```
Flipped the logic around, here.

mentisy replied:
You're right, that would make more sense. As you've committed the change, I'll set this as resolved.

ravage84 commented on src/Template/Element/status_table.ctp:
```diff
@@ -58,6 +59,10 @@ $sensorStatuses->some(function ($sensorStatus) {
             ],
+            [
+                $wasCheckFromCache ? 'Cached' : 'Not cached',
+                ['class' => $wasCheckFromCache ? 'primary' : 'secondary']
```
Where are these classes from?

mentisy replied:
They are bootstrap classes. The template file has previously used them in the table rows. See line 31-35.

mentisy replied:
The classes were from Bootstrap 4, but I see that the existing templates match with version 3 of Bootstrap. I've made changes to the classes so they use version 3 as well.

ravage84 commented on tests/TestCase/Heartbeat/SensorTest.php:
```diff
@@ -73,4 +73,37 @@ public function testGetStatus()
+     * Tests whether the check result was fetched from cache or by running the check now
+     *
+     * @return void
+     * @covers Sensor\Status::wasCheckCached
```
This test should be moved to StatusTest, I guess.

https://github.com/orca-services/cakephp-heartbeat/blob/9cfcd57597d9e589e0418a8e8d527263e28cf041/tests/TestCase/Heartbeat/Sensor/StatusTest.php

mentisy replied:
Moved the test.

ravage84 commented on tests/TestCase/Heartbeat/SensorTest.php:
```diff
@@ -73,4 +73,37 @@ public function testGetStatus()
+    public function testWasCheckCached()
+    {
+        Chronos::setTestNow('2017-03-30 12:45:37');
+        $sensorName = 'Dummy Sensor';
```
May be "Cached Sensor" is better?

mentisy replied:
To be honest, I just used the same name that other tests used. Changed name in a commit moving the test from your previous comment.

mentisy commented:
Thanks for your feedback, @ravage84. I'll have to amend the changes / get back to you on Wednesday.

ravage84 commented on src/Template/Element/status_table.ctp:
```diff
@@ -58,6 +59,10 @@ $sensorStatuses->some(function ($sensorStatus) {
             ],
+            [
+                $wasCheckFromCache ? 'Cached' : 'Not cached',
+                ['class' => $wasCheckFromCache ? 'success' : 'info']
```
Not sure these classes make sense in that case....

mentisy replied:
You're probably right. They are mostly decorative, and when the decoration is unclear, there's no point. Should I commit a change without assigning a class to the column.

ravage84 commented:
@mentisy thanks a lot for your contribution. I improved it and fixed a cache config bug alongside.

Could you please have alook at the new change set?

mentisy commented:
> @mentisy thanks a lot for your contribution. I improved it and fixed a cache config bug alongside.
> 
> Could you please have alook at the new change set?

Looks good. I also removed the decorative css classes from the template file displaying cached status. As you said, they didn't make much sense.

liviakuenzli commented:
Thank you for your contribution! 💛 💛 💛
